hipcheck/src/metric/entropy.rs

```diff
@@ -158,52 +158,62 @@ fn grapheme_freqs(commit_diff: &CommitDiff, db: &dyn MetricProvider) -> Result<C
+	let tgt_diffs: Result<Vec<&FileDiff>> = commit_diff
+		.diff
+		.file_diffs
+		.iter()
+		.filter_map(|file_diff| {
+			// Filter out any that are probably not source files, or are empty patches
+			let is_source: bool = match db.is_likely_source_file(Arc::clone(&file_diff.file_name)) {
+				Err(e) => {
+					return Some(Err(e));
+				}
+				Ok(s) => s,
+			};
+			if !is_source || file_diff.patch.is_empty() {
+				None
+			} else {
+				Some(Ok(file_diff))
+			}
+		})
+		.collect();
+	tgt_diffs?
+		.par_iter()
+		.map(|file_diff| {
+			// Count the number of graphemes in this patch, add it to the total,
+			// and track the number of each grapheme.
+			file_diff
+				.patch
+				// Iterate in parallel over the lines of the patch.
+				.par_lines()
+				// Normalize each line.
+				// See https://en.wikipedia.org/wiki/Unicode_equivalence.
+				.map(|line: &str| line.chars().nfc().collect::<String>())
+				// Count the graphemes in each normalized line.
+				// Also update the graphemes table here.
+				// We'll sum these counts to get the total number of graphemes.
+				.map(|normalized_line: String| {
+					// Create an iterator over the graphemes in the line.
+					Graphemes::new(&normalized_line)
+						// Update the graphemes table.
+						.map(|grapheme: &str| {
+							// Use this if statement to avoid allocating a new string unless needed.
+							if let Some(mut count) = grapheme_table.get_mut(grapheme) {
+								*count += 1;
+							} else {
+								grapheme_table.insert(grapheme.to_owned(), 1);
+							}
+						})
+						// get the grapheme count for this normalized line.
+						.count()
+				})
+				.sum::<usize>()
+		})
+		.collect_into_vec(&mut res);
 
-		// Count the number of graphemes in this patch, add it to the total,
-		// and track the number of each grapheme.
-		total_graphemes += file_diff
-			.patch
-			// Iterate in parallel over the lines of the patch.
-			.par_lines()
-			// Normalize each line.
-			// See https://en.wikipedia.org/wiki/Unicode_equivalence.
-			.map(|line: &str| line.chars().nfc().collect::<String>())
-			// Count the graphemes in each normalized line.
-			// Also update the graphemes table here.
-			// We'll sum these counts to get the total number of graphemes.
-			.map(|normalized_line: String| {
-				// Create an iterator over the graphemes in the line.
-				Graphemes::new(&normalized_line)
-					// Update the graphemes table.
-					.map(|grapheme: &str| {
-						// Use this if statement to avoid allocating a new string unless needed.
-						if let Some(mut count) = grapheme_table.get_mut(grapheme) {
-							*count += 1;
-						} else {
-							grapheme_table.insert(grapheme.to_owned(), 1);
-						}
-					})
-					// get the grapheme count for this normalized line.
-					.count()
-			})
-			.sum::<usize>();
-	}
+	// Use this variable to track the total number of graphemes accross all patches in this commit diff.
+	let total_graphemes: usize = res.into_iter().sum();
```

Comment:
Instead of `collect_into_vec` and then turning the `res` vec into an iterator and summing, we should just sum directly here (and avoid allocating a `res` vec at all).